proposals/native-integers.md

```diff
@@ -0,0 +1,342 @@
+| `nint` | `object` | Boxing | `box` |
+| `nint` | `void*` | PointerToVoid | `conv.i` |
+| `nint` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `nint` | `sbyte` | ExplicitNumeric | `conv.i1` / `conv.ovf.i1` |
+| `nint` | `byte` | ExplicitNumeric | `conv.u1` / `conv.ovf.u1` |
+| `nint` | `short` | ExplicitNumeric | `conv.i2` / `conv.ovf.i2` |
+| `nint` | `ushort` | ExplicitNumeric | `conv.u2` / `conv.ovf.u2` |
+| `nint` | `int` | ExplicitNumeric | `conv.i4` / `conv.ovf.i4` |
+| `nint` | `uint` | ExplicitNumeric | `conv.u4` / `conv.ovf.u4` |
+| `nint` | `long` | ImplicitNumeric | `conv.i8` / `conv.ovf.i8` |
+| `nint` | `ulong` | ExplicitNumeric | `conv.i8` / `conv.ovf.i8` |
+| `nint` | `char` | ExplicitNumeric | `conv.u2` / `conv.ovf.u2` |
+| `nint` | `float` | ImplicitNumeric | `conv.r4` |
+| `nint` | `double` | ImplicitNumeric | `conv.r8` |
+| `nint` | `decimal` | ImplicitNumeric | `conv.i8 decimal decimal.op_Implicit(long)` |
+| `nint` | `IntPtr` | Identity | |
+| `nint` | `UIntPtr` | None | |
+| `nuint` | `object` | Boxing | `box` |
+| `nuint` | `void*` | PointerToVoid | `conv.u` |
+| `nuint` | `nint` | ExplicitNumeric | `conv.i` / `conv.ovf.i` |
+| `nuint` | `sbyte` | ExplicitNumeric | `conv.i1` / `conv.ovf.i1` |
+| `nuint` | `byte` | ExplicitNumeric | `conv.u1` / `conv.ovf.u1` |
+| `nuint` | `short` | ExplicitNumeric | `conv.i2` / `conv.ovf.i2` |
+| `nuint` | `ushort` | ExplicitNumeric | `conv.u2` / `conv.ovf.u2` |
+| `nuint` | `int` | ExplicitNumeric | `conv.i4` / `conv.ovf.i4` |
+| `nuint` | `uint` | ExplicitNumeric | `conv.u4` / `conv.ovf.u4` |
+| `nuint` | `long` | ExplicitNumeric | `conv.i8` / `conv.ovf.i8` |
+| `nuint` | `ulong` | ImplicitNumeric | `conv.u8` / `conv.ovf.u8` |
+| `nuint` | `char` | ExplicitNumeric | `conv.u2` / `conv.ovf.u2.un` |
+| `nuint` | `float` | ImplicitNumeric | `conv.r.un conv.r4` |
+| `nuint` | `double` | ImplicitNumeric | `conv.r.un conv.r8` |
+| `nuint` | `decimal` | ImplicitNumeric | `conv.u8 decimal decimal.op_Implicit(ulong)` |
+| `nuint` | `IntPtr` | None | |
+| `nuint` | `UIntPtr` | Identity | |
+
+Conversion from `A` to `Nullable<B>` is:
+- an implicit nullable conversion if there is an identity conversion or implicit conversion from `A` to `B`;
+- an explicit nullable conversion if there is an explicit conversion from `A` to `B`;
+- otherwise invalid.
+
+Conversion from `Nullable<A>` to `B` is:
+- an explicit nullable conversion if there is an identity conversion or implicit or explicit numeric conversion from `A` to `B`;
+- otherwise invalid.
+
+Conversion from `Nullable<A>` to `Nullable<B>` is:
+- an identity conversion if there is an identity conversion from `A` to `B`;
+- an explicit nullable conversion if there is an implicit or explicit numeric conversion from `A` to `B`;
+- otherwise invalid.
+
+### Operators
+
+The predefined operators are as follows.
+These operators are considered during overload resolution based on normal rules for implicit conversions of arguments.
```

Comment:
>based on normal rules for implicit [](start = 58, length = 34)

These operators are only considered if one of the operands is of type `nint` or `nuint`.